libcpu/risc-v/common64/asid.c

```diff
@@ -29,9 +29,10 @@ static rt_uint64_t global_asid_generation;
 
 void rt_hw_asid_init(void)
 {
-    unsigned int satp_reg = read_csr(satp);
+    rt_uint64_t satp_reg = read_csr(satp);
     satp_reg |= (((rt_uint64_t)0xffff) << PPN_BITS);
     write_csr(satp, satp_reg);
+    asm volatile("sfence.vma x0, x0");
```

Comment:
建议使用 `__asm__`，对非 GNU 的 C compiler 更友好一些。

```suggestion
    __asm__ volatile("sfence.vma x0, x0");
```

另外感觉这里立即重置 satp 怪怪的。因为正常是两个启动阶段，做两次 satp 设置。第一次是初始映射，使用大粒度 pte，为了在初始化中设置最基础的虚存属性，并进入线性地址空间。第二次是按照虚存配置的映射，可能从静态/动态配置中读取运行时虚存布局，按照应用、驱动的需求配置内核地址空间。

这里在中间又重置了一次 satp，并不是特别必要。或许是可以把 ASID 初始化操作合并到 `rt_hw_mem_setup_early` 中。

Reply:
这里主要是为了获取mmu asid的信息，因为涉及到写satp寄存器，确实合并到rt_hw_mem_setup_early 中会更合理些